### DISCUSSION TITLE
W m^2 in a photometric passband

### DISCUSSION THREAD

**Discussion by jrthorstensen** [phoebe-project/phoebe2#492] [Questions]
Hello -- 
I'm modeling a light curve for a system with a good Gaia distance, and for which I can make reasonable guesses as to the radii and effective temperatures, so any physical model predicts an actual flux at earth.  I have the ZTF light curve, which is of course in g and r magnitudes, as well as some 'white light' light curves that I can interpret crudely, at least.  I've managed to generate a light curve that has fairly realistic values of flux, by setting the radii and effective temperatures, downloaded and set an r-passband

b.add_dataset('lc', times=times, dataset='r',passband = 'PanStarrs:r')

and exposed the absolute fluxes:

b.set_value('pblum_mode','absolute')

run the model,

b.run_compute(model = 'nominal')

and the numbers in 

b['value@fluxes@r@nominal@model']

are said to be in units of W/m^2, and are roughly correct for the flux integrated -- somehow! -- over the passband.

The last hurdle is -- how to compare these to magnitudes?  The flux is not given as f-lambda or f-nu, so there's some kind of factor involving the passband width (and maybe even the shape, if we're fussy) for which I can't find documentation.  Any suggestions?  If there's a crude way with a constant, so that e.g. V = -2.5 log_10 (phoebe flux) + constant, that would probably be good enough, since this is a bit rough anyway.

It's possible that what I should be doing is using phoebe's own provisions for matching the computed light curve to observations, but I'm not yet clear as to how to do that.  I'm finding that as advertised, the learning curve is steep, and if I might suggest a remedy, it would be to write an overall narrative guide discussing the overall conceptual framework and terminology, for example, describing exactly what's meant a 'dataset' in the phoebe world.  Phoebe is an astonishingly powerful and beautifully-crafted program, but I think the potential barrier around it is higher than it needs to be.

Thanks - John 

[This seems related to previous questions by dai799 on April 27 and podesse on March 16, but it's a bit different.]

**Comment by aprsa** [comment]
Hi John,

phoebe works exclusively in fluxes, so any conversion to magnitudes will have to be done outside of phoebe. What you really need is the magnitude zero-point, which is in the realm of photometric passbands (see, for example, [IAU resolution B2](https://arxiv.org/abs/1510.06262) from 2015). The treatment of atmospheres is concisely described in [Prsa et al. (2016)](https://ui.adsabs.harvard.edu/abs/2016ApJS..227...29P/abstract). If you want the full gory detail, it's in [Prsa (2018)](https://ui.adsabs.harvard.edu/abs/2018maeb.book.....P/abstract).

To achieve what you want solely in phoebe, you might want to create a "calibration" star -- i.e. a single star with a known temperature, size, distance, compute its flux in a passband that you want to calibrate, and then use that flux as calibration flux in the flux-magnitude equation.

Under the hood phoebe includes pre-integrated tables of f-lambda SEDs (specific intensities, so W/m^3 rather than W/m^2) multiplied by the passband response function (in both energy-weighted and photon-weighted mode). But if you wanted to integrate SEDs yourself and derive zero-mag points yourself, that can be done too -- let me know if you decide to go that route and I'll do my best to help. 

Just note that I'm currently diving in Key Largo so will be slower to respond.

**Reply by jrthorstensen** [reply]
Thanks!  I think I'll try your suggestion of a 'calibration' star and see how it goes.  Have fun in Key Largo!

**Reply by jrthorstensen** [reply]
As a followup, I did just that -- created a solar-type star at 10 pc, looked up the absolute magnitude in the passband I wanted from Willmer 2018 ApJS 236 47, and used that to find the constant.  This of course ignores any effects of the finite filter width, but this is far from precise work.
